modules/etcd-creating-automated-backups.adoc

```diff
@@ -70,11 +70,108 @@ Follow these steps to create a single etcd backup by creating and applying a cus
 
 * You have access to the cluster as a user with the `cluster-admin` role.
 * You have access to the OpenShift CLI (`oc`).
-* You have a PVC to save backup data to.
 
 .Procedure
 
-. Create a CR file named `etcd-single-backup.yaml` with contents such as the following example:
+. Create a local storage class by completing the following steps:
```

Comment:
Thanks @lahinson for the update 

So what I meant is to clarify the `PVC` prerequisite to the user 

- The user needs a PVC before creating the CR as a prerequisite
- The PVC could be one of two things, dynamically provisioned or manually provisioned 


For dynamically provisioned PVC :- 

- This exist on clusters with cloud providers (e.g. aws, gcp, azure). 
- This is being already documents in the current online docs, however the `storageClass` name used is no longer correct, at least on 4.16 cluster i used. Please verify this with the cc @openshift/storage.
```
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: etcd-backup-pvc
  namespace: openshift-etcd
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 200Gi 
  storageClassName: standard-csi 
  volumeMode: Filesystem
```
- In case the cluster does not provide `dynamic provisioning` or the user does not want to use it, the other option is to use create `StorageClass`, `PV`, and `PVC` as explained in the doc I shared.

My comments on current docs 

- the way it explains the PVC is not clear 
- the current PVC inlined in the docs relied on a storage class that no longer exist as of 4.16 `storageClassName: standard-csi`
- the other approach of using a `localStorageClass` will never work, since the name of the storageClass being used in the current doc is incorrect 

```
kind: StorageClass
metadata:
  name: etcd-backup-local-storage
```

```
 persistentVolumeReclaimPolicy: Delete
  storageClassName: local-storage
```

```
volumeMode: Filesystem
  resources:
    requests:
      storage: 10Gi 
  storageClassName: local-storage
```

Reply:
@Elbehery Thanks for the clarification! If `standard-csi` no longer exists and `localStorageClass` won't work, what should I specify for the storage class?

Reply:
so each OCP cluster has a `default` storage class which is being used by default if you do **omit** specify any `storageClassName` field from the PVC manifest. Note that this applies only to OCP clusters which supports CSI - usually CSI exist on OCP cluster backed by cloud providers as listed [here](https://docs.openshift.com/container-platform/4.15/storage/dynamic-provisioning.html).

example 

```
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: etcd-backup-pvc
  namespace: openshift-etcd
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 200Gi 
  volumeMode: Filesystem
```

Again this needs to be confirmed with @openshift/storage to avoid mistakes.

Reply:
`localStorageClass` **does** work, but the issue in the manifest .. the name of the storage class used was not identical to the defined storage class. 

You can use the following manifest for `localStorageClass` I have tested them already 

```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: etcd-backup-local-storage
provisioner: kubernetes.io/no-provisioner
volumeBindingMode: Immediate

---

apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: etcd-backup-pvc
  namespace: openshift-etcd
spec:
  accessModes:
  - ReadWriteOnce
  volumeMode: Filesystem
  resources:
    requests:
      storage: 10Gi 
  storageClassName: etcd-backup-local-storage

---

apiVersion: v1
kind: PersistentVolume
metadata:
  name: etcd-backup-pv-fs
spec:
  capacity:
    storage: 100Gi 
  volumeMode: Filesystem
  accessModes:
  - ReadWriteOnce
  persistentVolumeReclaimPolicy: Retain
  storageClassName: etcd-backup-local-storage
  local:
    path: /mnt
  nodeAffinity:
    required:
      nodeSelectorTerms:
      - matchExpressions:
        - key: kubernetes.io/hostname
          operator: In
          values:

---

apiVersion: operator.openshift.io/v1alpha1
kind: EtcdBackup
metadata:
  name: etcd-single-backup
  namespace: openshift-etcd
spec:
  pvcName: etcd-backup-pvc
```